LottieSample/src/androidTest/java/com/airbnb/lottie/samples/LottieIdlingAnimationResource.kt

```diff
@@ -0,0 +1,37 @@
+package com.airbnb.lottie.samples
+
+import android.animation.Animator
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.IdlingResource
+import com.airbnb.lottie.LottieAnimationView
+
+class LottieIdlingAnimationResource(animationView: LottieAnimationView?, private val name: String = "Lottie") : IdlingResource {
+
+    init {
+        animationView?.addAnimatorListener(object : AnimatorListenerAdapter() {
+            override fun onAnimationStart(animation: Animator) {
+                isIdle = false
+            }
+
+            override fun onAnimationEnd(animation: Animator) {
+                isIdle = true
+                callback?.onTransitionToIdle()
+                animationView.removeAllAnimatorListeners()
+                IdlingRegistry.getInstance().unregister(this@LottieIdlingAnimationResource)
+            }
+        })
+    }
+
+    private var callback: IdlingResource.ResourceCallback? = null
+    private var isIdle = animationView?.isAnimating?.not() ?: true
```

Comment:
I think this should be:
`private val isIdle get() = animationView?.isAnimating?.not() ?: true`
Then yiou don't need to set it on lines 13 or 17

Reply:
In such case we need to hold reference to `animationView: LottieAnimationView?`, make it val. We should decide between referencing a view directly and having a bit more code :) I don't really like referencing, but I'm almost sure it won't affect anything in current scenario.